service/serviceUtils.go

```diff
@@ -0,0 +1,205 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Informasjonsforvaltning/fdk-harvest-admin/config/env"
+	"github.com/Informasjonsforvaltning/fdk-harvest-admin/model"
+)
+
+func unmarshalAndValidateDataSource(bytes []byte) (*model.DataSource, error) {
+	var dataSource model.DataSource
+	err := json.Unmarshal(bytes, &dataSource)
+	if err != nil {
+		return nil, err
+	}
+
+	err = dataSource.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	return &dataSource, nil
+}
+
+func dataTypeToMessageKey(dataType model.DataTypeEnum) (*string, error) {
+	switch dataType {
+	case model.Concept:
+		msgKey := messageKey("concept")
+		return &msgKey, nil
+	case model.Dataset:
+		msgKey := messageKey("dataset")
+		return &msgKey, nil
+	case model.InformationModel:
+		msgKey := messageKey("informationmodel")
+		return &msgKey, nil
+	case model.DataService:
+		msgKey := messageKey("dataservice")
+		return &msgKey, nil
+	case model.PublicService:
+		msgKey := messageKey("publicservice")
+		return &msgKey, nil
+	}
+
+	return nil, fmt.Errorf("%s is not a valid data type", dataType)
+}
+
+func messageKey(messageType string) string {
+	return fmt.Sprintf("%s.%s.%s", messageType, env.ConstantValues.RabbitMsgKeyMiddle, env.ConstantValues.RabbitMsgKeyEnd)
+}
+
+func HarvestTypeFromRoutingKey(routingKeyPrefix string) (*string, error) {
+	switch routingKeyPrefix {
+	case "concepts":
```

Comment:
Kan vurdere å bruke konstanten her, så man unngår muligheten for en skrivefeil.

Reply:
`routing key` og `harvest type` er ikke helt like, så her får du skrivefeil om du bruker konstantene. Skal jeg lage `RoutingKeyEnum` i  tillegg?

Reply:
Ah, ser nå at det er noen små forskjeller ja. Da kan du la det være :sweat_smile: